scripts/conp_to_nidm_terms/functions.py

```diff
@@ -0,0 +1,181 @@
+        key = api_key_json["api_key"] if api_key_json["api_key"] != "" else _raise_error(f"{api_key_json['_comment']}")
+
+    try:
+        api_key = f"?key={key}"
+        r = requests.get(NIF_API_URL + term + api_key, headers={'accept': 'application/json'})
+        r.raise_for_status()
+        response = json.loads(r.content.decode('utf-8'))
+        match = str()
+        # Standard response will have existing_ids key
+        if response["data"]["existing_ids"]:
+            for i in response["data"]["existing_ids"]:
+                # retrieve InterLex id, its curie has "ILX" prefix
+                match = i["iri"] if "curie" in i and "ILX:".upper() in i["curie"] else match
+        else:
+            match = "no match found"
+        return match
+
+    except requests.exceptions.HTTPError as e:
+        logger.error(f"Error: {e}")
+
+
+def collect_values(privacy=True, types=True, licenses=True, is_about=True, formats=True, keywords=True):
+    """
+    Iterates over projects directory retrieving DATS file for each project.
+    Aggregates all values and their count for selected properties in the report object.
+    :param : set to False in order not to include the property in the final report
+    :return: dict object report, int how many DATS files were processed
+    """
+
+    # Text values to collect
+    privacy_values = set()
+    licenses_values = set()
+    types_datatype_values = set()
+    is_about_values = set()
+    distributions_formats = set()
+    keywords_values = set()
+
+    dats_files_count = 0
+
+    # Access DATS.json in each project's root directory
+    for path, directories, files in os.walk(PROJECTS):
+        if "DATS.json" in files:
+            dats_files_count += 1
+            dats_file = os.path.join(path, "DATS.json")
+            with open(dats_file, "r", encoding="utf-8") as json_file:
+                dats_data = json.load(json_file)
+
+                # privacy is not required
+                if privacy and "privacy" in dats_data:
+                    privacy_values.add(dats_data["privacy"])
+
+                if types:
+                    # types are required
+                    for typ in dats_data["types"]:
+                        # types takes four possible datatype schemas
+                        datatype_schemas = ["information", "method", "platform", "instrument"]
+                        types_datatype_values.update(set(typ[t]["value"] for t in datatype_schemas if t in typ))
+
+                if licenses:
+                    # licenses is required
+                    licenses_values.update(set(l["name"] for l in dats_data["licenses"]))
+
+                # isAbout is not required
+                if is_about and "isAbout" in dats_data:
+                    for each_is_about in dats_data["isAbout"]:
+                        if "name" in each_is_about:
+                            is_about_values.add(each_is_about["name"])
+                        elif "value" in each_is_about:
```

Comment:
I'm confused. IsAbout can have both `name` and `value`? Is that expected?

Also, in here, you could maybe could skip the terms that have an identifier, like species who are related to NCBI taxonomy terms?

Reply:
`isAbout` can take nine different schemas inside, all of them have `name `required except for annotation_schema - the string value is in `value` there, so I am checking for it.


> Also, in here, you could maybe could skip the terms that have an identifier, like species who are related to NCBI taxonomy terms?

Do you mean the script shouldn't create a jsonld file for the species terms?

Reply:
@zxenia I don't know. Do the species need to be added to the NIDM terms if they are terms coming from the NCBI taxonomy database? I honestly don't know.